src/ray/core_worker/transport/direct_actor_task_submitter.cc

```diff
@@ -307,6 +296,17 @@ void CoreWorkerDirectActorTaskSubmitter::DisconnectActor(
     }
   }
 
+  if (!wait_for_death_info_tasks.empty()) {
+    auto error_type = GenErrorTypeFromDeathCause(death_cause);
+    const auto error_info = GetErrorInfoFromActorDeathCause(death_cause);
+    RAY_LOG(INFO) << "Failing tasks waiting for death info, size="
+                  << wait_for_death_info_tasks.size() << ", actor_id=" << actor_id;
+    for (auto &net_err_task : wait_for_death_info_tasks) {
+      RAY_UNUSED(task_finisher_.MarkTaskReturnObjectsFailed(
```

Comment:
so this will try to hold the lock again?

Reply:
Well, it won't, but the callback inside might. Check my test, I put `_on_completed` there and that function might.
It'll be set in async env, but i'm not sure how to construct a test case triggering this in async env.

Reply:
To be more specific, for this test case, it'll call submit task and submit task is trying to hold the lock. maybe we should use reentrant lock here.